app_server/app/controllers/AdminController.js

```diff
@@ -288,12 +305,21 @@ var encrypt = function(password) {
 };
 
 var updateCache = function(request, account, callback) {
-  request.server.app.cache.set(account.userId, account, 0, function (err) {
-    if (err) {
-      logger.error(err);
+  request.server.app.cache.get(account.userId, function(err, cached) {
+    if (cached && cached.item) {
+      // merge with cached version, with new taking priority
+      Object.keys(cached.item)
+            .filter((k) => !account.hasOwnProperty(k))
```

Comment:
I think  `Object.keys` already enumerate object's own properties only so you can choose to remove the `filter` function.